core/src/main/java/bio/terra/pearl/core/service/workflow/EnrollmentService.java

```diff
@@ -131,25 +142,43 @@ public HubResponse<Enrollee> enroll(String portalShortcode, EnvironmentName envN
     }
 
     /**
-     * Will create an enrollee for both the given user and a new governed user that is registered.  The HubResponse
-     * returned will include the governedEnrollee as the "response", and the enrollee corresponding to the proxy user as
-     * the "enrollee"
+     * Will check if proxy is already enrolled, if not will create an enrollee for the proxy. Will then create a new
+     * governed user that is registered.  The HubResponse returned will include the governedEnrollee as the "response",
+     * and the enrollee corresponding to the proxy user as the "enrollee"
      */
     @Transactional
-    public HubResponse<Enrollee> enrollAsProxy(String portalShortcode, EnvironmentName envName, String studyShortcode, ParticipantUser user, PortalParticipantUser ppUser, UUID preEnrollResponseId) {
-        HubResponse<Enrollee> proxyResponse = enroll(portalShortcode, envName, studyShortcode, user, ppUser, preEnrollResponseId, false);
-        HubResponse<Enrollee> governedResponse = enrollGovernedUser(portalShortcode, envName, studyShortcode, proxyResponse.getEnrollee(), user, ppUser, preEnrollResponseId);
-        governedResponse.setEnrollee(proxyResponse.getEnrollee());
+    public HubResponse<Enrollee> enrollAsProxy(EnvironmentName envName, String studyShortcode, ParticipantUser proxyUser,
+                                               PortalParticipantUser ppUser, UUID preEnrollResponseId) {
+        String governedUsernameSuffix = registrationService.generateGovernedUsernameSuffix(proxyUser.getUsername(), proxyUser.getEnvironmentName());
+        String governedUserName = GOVERNED_USERNAME_INDICATOR.formatted(proxyUser.getUsername(), governedUsernameSuffix);//a@b.com-prox-guid
+        return enrollAsProxy(envName, studyShortcode, proxyUser, ppUser, preEnrollResponseId, governedUserName);
+    }
+
+    @Transactional
+    public HubResponse<Enrollee> enrollAsProxy(EnvironmentName envName, String studyShortcode, ParticipantUser proxyUser,
+                                               PortalParticipantUser ppUser, UUID preEnrollResponseId, String governedUsername) {
+        Optional<Enrollee> maybeProxyEnrollee =
```

Comment:
The convention elsewhere in Juniper is to use `opt` to suffix optional local variables
```suggestion
        Optional<Enrollee> proxyEnrolleeOpt =
```
but see my comment below about how we should probably remove this local variable altogether